### PR TITLE
Improvements to speech-to-function (R11)

### PR DESCRIPTION
**Pull request by chauler**
# Changes

Updated server.py and nlp.py to allow for matching user input to commands that aren't natural english. For example, "Insert an if ladder" used to ignore the word if and not execute the desired command as a result.

# Testing (Windows Only)

On VS Code, execute the "Run Extension" debug profile.
On the newly opened debug window, in the Action List, click on "Voice-command.start Streaming".
Upon receiving the "Please give your command. Listening..." notification, say the title of a command provided by Mind Reader (titles are what show up in the Access Menu on the left sidebar, for example)
Enter a python file and say one of the code-related functions (e.g. insert an if ladder, insert a for loop, insert a try ladder)
Say "Exit" to stop the server from listening.

### CODE REVIEW

File: server.py
Comment (Kaiyan-Da-Man) [inline server.py:1]: While you're here, might as well fix the typo on line 76. I think 'Sever' is supposed to be 'Server'

**Comment by Kaiyan-Da-Man** [comment]
image

getting a timeout error. Is this error the intended behavior?

**Comment by Kaiyan-Da-Man** [comment]
image
terminal opens when I press `Voice-command.start Streaming`. Any way to get rid of this?

**Comment by Kaiyan-Da-Man** [comment]
Partially my fault, but I think we should remove the portaudio dependency for now since we have a solution for PyAudio. We can always add it back when we need it.

**Review by Kaiyan-Da-Man — CHANGES_REQUESTED** [review]

File: server.py
Comment (chauler) [reply server.py:1]: Fixed. Also removed PyAudio and removed the terminal window from server startup. The timeout error is unintended, but is known and part of Sprint 3's plans.

**Review by Kaiyan-Da-Man — APPROVED** [review]
lgtm

**Pull request merged**